timemachine/fe/atom_mapping.py

```diff
@@ -347,10 +351,18 @@ def chiral_filter(trial_core):
         passed = not has_chiral_atom_flips(np_core, chiral_set_a, chiral_set_b)
         return passed
 
+    def ring_matches_ring_filter(trial_core):
+        return all(
+            mol_a.GetAtomWithIdx(int(idx_a)).IsInRing() == mol_b.GetAtomWithIdx(int(idx_b)).IsInRing()
+            for idx_a, idx_b in enumerate(trial_core)
+            if idx_b != mcgregor.UNMAPPED
+        )
+
+    filter_fxn = lambda _: True
     if enforce_chiral:
-        filter_fxn = chiral_filter
-    else:
-        filter_fxn = lambda core: True
+        filter_fxn = compose_predicates(filter_fxn, chiral_filter)
+    if ring_matches_ring:
```

Comment:
Rather than doing it during the mcgregor algorithm, would it make more sense to just set up the priorities idxs to exclude two atoms that aren't both in a ring?

https://github.com/proteneer/timemachine/blob/master/timemachine/fe/atom_mapping.py#L312-L336

Reply:
Ah, yeah, that seems better. Will update.

Reply:
Simplified to enforce in the codomains in https://github.com/proteneer/timemachine/pull/1051/commits/6d6a0cddea5eaebc5cb29a2b7d937e594c20b194

Reply:
Nice -- restoring `ring_matches_ring_only` option is important, and applying this in predicates matrix / codomain structure is my implementation preference as well https://github.com/proteneer/timemachine/pull/921 (not sure why I didn't finish this one)

Reply:
Oh wow, I hadn't seen #921. I cherry-picked the docstring and some further improvements from there; the implementation in this PR is nearly identical now.